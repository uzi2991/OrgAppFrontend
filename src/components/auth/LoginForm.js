import React, { useContext } from 'react';
import axios from 'axios';

import globalContext from '../../context/globalContext';
import { backendUrl } from '../../static/js/const';
import { useForm } from 'react-hook-form';

const LoginForm = ({ setErrMsgs }) => {
  const { register, handleSubmit, watch } = useForm();

  const email = watch('email', '');
  const password = watch('password', '');
  const { login } = useContext(globalContext);

  const onSubmit = async (data) => {
    const url = `${backendUrl}/auth/signin`;
    try {
      const res = await axios.post(url, data);
      login(res.data);
    } catch (err) {
      if (err.response?.status === 401) {
        setErrMsgs({
          signup: false,
          err: true,
          msgs: { Invalid: 'username or password' },
        });
      }
    }
  };

  return (
    <div className="login">
      <form onSubmit={handleSubmit(onSubmit)}>
        <fieldset className="login-fieldset">
          <input
            className="sidebar-input border--gray border--onHoverBlue"
            type="text"
            name="email"
            placeholder="Email"
            ref={register({ required: true })}
          />
          <input
            className="sidebar-input border--gray border--onHoverBlue"
            type="password"
            name="password"
            placeholder="Password"
            ref={register({ required: true })}
          />
          {email.trim() !== '' && password.trim() !== '' ? (
            <button className="btn" type="submit">
              {' '}
              Login{' '}
            </button>
          ) : (
            <button className="btn btn--disabled" disabled>
              {' '}
              Login{' '}
            </button>
          )}
        </fieldset>
      </form>
    </div>
  );
};

export default LoginForm;
