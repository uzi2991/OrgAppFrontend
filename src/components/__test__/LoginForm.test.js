import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import LoginForm from "../auth/LoginForm";

test("LoginForm snapshot", () => {
    const loginFormSnapshot = renderer.create(
        <LoginForm />
    ).toJSON();

    expect(loginFormSnapshot).toMatchSnapshot();
})

