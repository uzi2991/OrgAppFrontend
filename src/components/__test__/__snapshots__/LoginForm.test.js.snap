// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot 1`] = `
<div
  className="login"
>
  <form
    onSubmit={[Function]}
  >
    <fieldset
      className="login-fieldset"
    >
      <input
        className="sidebar-input border--gray border--onHoverBlue"
        name="email"
        placeholder="Email"
        type="text"
      />
      <input
        className="sidebar-input border--gray border--onHoverBlue"
        name="password"
        placeholder="Password"
        type="password"
      />
      <button
        className="btn btn--disabled"
        disabled={true}
      >
         
        Login
         
      </button>
    </fieldset>
  </form>
</div>
`;
